chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -243,27 +243,37 @@ out:
 					"subscription: txid=%v, numconfs=%v",
 					msg.TxID, msg.NumConfirmations)
 
-				_, currentHeight, err := b.chainConn.GetBestBlock()
-				if err != nil {
-					chainntnfs.Log.Error(err)
-				}
+				currentHeight := uint32(bestHeight)
+
+				// Look up whether the transaction is already
+				// included in the active chain. We'll do this
+				// in a goroutine to prevent blocking
+				// potentially long rescans.
+				go func() {
+					confDetails, err := b.historicalConfDetails(
+						msg.TxID, msg.heightHint,
+						currentHeight,
+					)
+					if err != nil {
+						chainntnfs.Log.Error(err)
+						return
+					}
 
-				// Lookup whether the transaction is already included in the
-				// active chain.
-				txConf, err := b.historicalConfDetails(
-					msg.TxID, msg.heightHint, uint32(currentHeight),
-				)
-				if err != nil {
-					chainntnfs.Log.Error(err)
-				}
+					if confDetails != nil {
+						err := b.txConfNotifier.UpdateConfDetails(
+							*msg.TxID, msg.ID,
```

Comment:
I think it's better to keep it as is to make sure the notification has been registered with the notifier first.

Reply:
if moving the `Register` call right before, that would make those call more tightly coupled, and i think even the whole `ConfID` can be removed?

Reply:
Sure, but then it's expected for the caller to know this, rather than just providing an interface that ensures the notification has been registered first. Having an ID will also come in handy when adding support for cancelling existing notifications (there's an existing TODO for this).